libs/utils/src/lib/command-runner.service.ts

```diff
@@ -65,17 +78,90 @@ export class CommandRunner {
       );
   }
 
-  stopCommand() {
-    this.runCommand(
-      gql`
-        mutation {
-          stop {
-            result
+  listAllCommands(): Observable<CommandResponse[]> {
+    return this.apollo
+      .watchQuery({
+        pollInterval: COMMANDS_POLLING,
+        query: gql`
+          {
+            commands {
+              id
+              status
+              workspace
+              command
+            }
+          }
+        `
+      })
+      .valueChanges.pipe(map((r: any) => r.data.commands));
+  }
+
+  getCommand(id: string): Observable<CommandResponse> {
+    return this.apollo
+      .watchQuery({
+        query: gql`
+          query($id: String) {
+            commands(id: $id) {
+              id
+              workspace
+              command
+              status
+              out
```

Comment:
If performance becomes an issue we should make it such that we only pull to get the full output on the first emission. Can you leave a comment here to that effect?

Reply:
I'll leave a comment.